Automattic/wp-calypso
# Inject `subscriptionManagerSubkey`in currentUser

## Proposed Changes

This was originally in [this PR](https://github.com/Automattic/wp-calypso/pull/74657), but then Github did a bit wonky when I merged with trunk.

This PR adds  a `subscriptionManagementSubkey` key on the `currentUser` global when:
- A cookie named `subkey` is set
- The user is logged out

This key is then used by the new Subscription Management portal. Because said cookie is not transmitted to the API (CORS), and we can't read it with JavaScript (it's `httpOnly`) we do it this way. The alternative is to set httpOnly to false, but this would theoretically allow cookie stealing on WP.com subdomains.

## Testing Instructions

1. Apply this PR & restart Calypso
2. Make sure you're logged out, and navigate to an accessible Calypso page like `http://calypso.localhost:3000/subscriptions/settings`
3. Set a subkey cookie (either using Devtools), or by pasting `document.cookie = "subkey=test"` in the console.
4. Refresh & paste this in the console: `window.currentUser?.subscriptionManagementSubkey` - you should see `test`

## Pre-merge Checklist
